packages/forms/src/UIForm/Widget/condition.js

```diff
@@ -1,18 +1,91 @@
 import get from 'lodash/get';
-import includes from 'lodash/includes';
+import findIndex from 'lodash/findIndex';
 
-function getValue(value, strategy) {
-	if (!strategy) {
+function toNumber(value) {
+	if (typeof value === 'string') {
+		return Number(value);
+	}
+	return value;
+}
+
+function toString(value) {
+	if (!value) {
 		return value;
 	}
-	switch (strategy.toLowerCase()) {
+	if (typeof value === 'number') {
+		return JSON.stringify(value);
+	}
+	if (value.toString) {
+		return value.toString();
+	}
+	return value;
+}
+
+function parseStrategy(strategy) {
+	if (!strategy) {
+		return undefined;
+	}
+
+	const start = strategy.indexOf('(');
+	if (start > 0) {
+		const end = strategy.indexOf(')', start);
+		if (end > 0) {
+			const params = strategy
+				.substring(start + 1, end)
+				.split(';')
+				.map(it => {
+					const sep = it.indexOf('=');
+					if (sep > 0) {
+						return {
+							[it.substring(0, sep).trim()]: it.substring(sep + 1, it.length).trim(),
+						};
+					}
+					return {
+						value: it.trim(),
+					};
+				})
+				.reduce(
+					(a, v) => ({
+						...a,
+						...v,
+					}),
+					{},
+				);
+			return {
+				name: strategy.substring(0, start).toLowerCase(),
+				params,
+			};
+		}
+	}
+	return {
+		name: strategy.toLowerCase(),
+		params: {},
+	};
+}
+
+function toEvaluator(value, strategyConfig) {
+	switch (strategyConfig.name) {
 		case 'length':
 			if (value && value.length) {
-				return value.length;
+				const length = value.length;
+				return expected => length === toNumber(expected);
+			}
+			return expected => expected === 0 || expected === '0';
```

Comment:
`expected` should be normalized to a number so that you don't have to compare to `'0'`.

Reply:
Moreover : 0 is the default value against checking the expectation ? If so, define the value first (length) with a default (0) if not provided, then do the same check for all.